src/components/icon/icon.style.js

```diff
@@ -0,0 +1,136 @@
+  if (bgTheme !== 'none') {
+    if (disabled) return theme.icon.disabled;
+  }
+
+  const palette = generatePalette({ statusColor: theme.colors[bgTheme], businessColor: theme.colors.primary });
+  const statuses = ['info', 'error', 'success', 'warning'];
+  if (statuses.includes(bgTheme)) {
+    return isHover ? palette.statusColorShade(20) : theme.colors[bgTheme];
+  }
+
+  if (bgTheme === 'business') {
+    return isHover ? palette.businessColorShade(20) : theme.colors.primary;
+  }
+
+  return 'transparent';
+};
+
+const getIconColor = (bgTheme, theme, iconColor, disabled, isHover) => {
+  if (disabled) return theme.icon.disabled;
+
+  if (bgTheme !== 'none') {
+    const whiteIconBackgrounds = ['error', 'info', 'business'];
+    const darkIconBackgrounds = ['success', 'warning'];
+
+    if (whiteIconBackgrounds.includes(bgTheme)) return theme.colors.white;
+
+    if (darkIconBackgrounds.includes(bgTheme)) {
+      return isHover ? theme.icon.defaultHover : theme.icon.default;
+    }
+  }
+
+  const palette = generatePalette({ businessColor: theme.colors.primary });
+  switch (iconColor) {
+    case 'on-dark-background':
+      return theme.colors.white;
+    case 'on-light-background':
+      return isHover ? theme.icon.onLightBackgroundHover : theme.icon.onLightBackground;
+    case 'business-color':
+      return isHover ? palette.businessColorShade(20) : theme.colors.primary;
+    default:
+      return isHover ? theme.icon.defaultHover : theme.icon.default;
+  }
+};
+
+const StyledIcon = styled.span`
+  ${({
+    bgTheme, theme, iconColor, bgSize, bgShape, isFont, type, fontSize, disabled
+  }) => css`
+    display: inline-block;
+    position: relative;
+    color: ${getIconColor(bgTheme, theme, iconColor, disabled, false)};
+    background-color: ${getBackgroundColor(theme, bgTheme, disabled, false)};
+
+    &:hover {
+      color: ${getIconColor(bgTheme, theme, iconColor, disabled, true)};
+      background-color: ${getBackgroundColor(theme, bgTheme, disabled, true)};
+    }
+
+    ${bgTheme !== 'none' && css`
+        align-items: center;
+        display: inline-flex;
+        justify-content: center;
+        height: ${iconSizeConfig.backgroundSize[bgSize]};
+        width: ${iconSizeConfig.backgroundSize[bgSize]};
+        border-radius: ${iconSizeConfig.backgroundShape[bgShape]};
+      `}
+
+    ${isFont && css`
+      &::before {
+        -webkit-font-smoothing: antialiased;
+        -moz-osx-font-smoothing: grayscale;
+
+        font-family: CarbonIcons;
+        content: "${iconUnicodes[type]}";
+        font-size: ${iconSizeConfig.iconSize[fontSize]};
+        font-style: normal;
+        font-weight: normal;
+        line-height: ${iconSizeConfig.iconSize[fontSize]};
+        vertical-align: middle;
+      }
+    `}
+
+    ${classicIconStyles};
+  `}
+`;
+
+StyledIcon.propTypes = {
+  theme: PropTypes.object,
+  isFont: PropTypes.bool,
+  type: PropTypes.oneOf(OptionsHelper),
```

Comment:
Shouldn't this be `PropTypes.oneOf(OptionsHelper.SOMETHING)` instead of `PropTypes.oneOf(OptionsHelper)` ?

Reply:
Yep good spot, I've updated it and also updated it in `icon.component` as it was set to string for some reason

Reply:
:white_check_mark: